templates/base-pipeline/template.yml

```diff
@@ -0,0 +1,210 @@
+      JQ_EXIT_CODE=$?
+
+      if [ $JQ_EXIT_CODE -ne 0 ]; then
+        echo -e "\e[31mError: Failed to parse lorenz status file\e[0m"
+        echo "File content sample:"
+        head -n 5 "$LORENZ_STATUS_FILE" 2>/dev/null || echo "Cannot read file"
+        exit 1
+      fi
+
+      if [ "$NODES_UP" = "null" ]; then
+        echo -e "\e[31mError: Machine ${CI_MACHINE} not found in status file\e[0m"
+        echo "Available machines in status file:"
+        jq -r 'keys[]' "$LORENZ_STATUS_FILE" 2>/dev/null | head -10 || echo "Cannot list machines"
+        exit 1
+      fi
+
+      # Check if machine is available - report to GitHub
+      if [ "$NODES_UP" -eq 0 ]; then
+        echo -e "\e[31mMachine ${CI_MACHINE} is down (${NODES_UP} nodes up)\e[0m"
+
+        # Report machine unavailability to GitHub
+        if [ -n "$[[ inputs.github_token ]]" ] && [ -n "$[[ inputs.github_project_org ]]" ] && [ -n "$[[ inputs.github_project_name ]]" ]; then
+          STATUS_RESPONSE=$(mktemp)
+          STATUS_HTTP_CODE=$(curl --retry 3 --retry-delay 5 --max-time 30 \
+              --url "https://api.github.com/repos/$[[ inputs.github_project_org ]]/$[[ inputs.github_project_name ]]/statuses/${CI_COMMIT_SHA}" \
+              --header 'Content-Type: application/json' \
+              --header "authorization: Bearer $[[ inputs.github_token ]]" \
+              --data "{ \"state\": \"failure\", \"target_url\": \"${CI_PIPELINE_URL}\", \"description\": \"GitLab: ${CI_MACHINE} down (0 nodes)\", \"context\": \"ci/gitlab/${CI_MACHINE}\" }" \
+              --output "$STATUS_RESPONSE" \
+              --write-out "%{http_code}" \
+              --silent \
+              --show-error)
+
+          echo "GitHub Status API response code: $STATUS_HTTP_CODE"
+
+          if [ "$STATUS_HTTP_CODE" -eq 201 ]; then
+            echo "Successfully reported machine down status to GitHub"
+          else
+            echo "Warning: Failed to report status to GitHub. HTTP status: $STATUS_HTTP_CODE"
+            echo "Response body:"
+            cat "$STATUS_RESPONSE"
+          fi
+
+          rm -f "$STATUS_RESPONSE"
+        else
+          echo "Warning: GitHub reporting variables not set, skipping GitHub status update"
+        fi
+
+        exit 1
+      else
+        echo -e "\e[32m${CI_MACHINE} is available (${NODES_UP} nodes up)\e[0m"
+      fi
+
+# Template for triggering build-and-test child pipelines
+# Projects should extend this for each machine
+# Requires: CI_MACHINE variable to be set
+.build-and-test:
+  stage: build-and-test
+  trigger:
+    strategy: depend
+    forward:
+      pipeline_variables: true
+
+# Template for custom job definitions
+# Projects extend this in their custom-jobs-and-variables.yml
+.custom_job:
+  variables:
+    GIT_STRATEGY: recursive
+    GIT_SUBMODULE_STRATEGY: recursive
+
+# Template for custom performance job definitions
+.custom_perf:
+  variables:
+    GIT_STRATEGY: recursive
+    GIT_SUBMODULE_STRATEGY: recursive
```

Comment:
I think the machine/performance pipeline components don't see this here, so it might have to be duplicated in each component. Noticed this because I just [tested](https://lc.llnl.gov/gitlab/quantum1/quandary/-/pipelines/612852) with no .custom_perf on Quandary and got ".perf_on_dane: unknown keys in `extends` (.custom_perf)" (but the rest worked!)

Reply:
Done!